zerver/lib/html_diff.py

```diff
@@ -74,6 +74,20 @@ def verify_html(html):
         return False
     return True
 
+def check_tags(text):
+    # type: (Text) -> Text
+    # The current diffing algorithm produces malformed html when text is
+    # added to existing new lines. This patch manually corrects that.
+    in_tag = False
+    if text.endswith('<'):
+        text = text[:-1]
+    for c in text:
+        if c == '<':
+            in_tag = True
+        elif c == '>' and not in_tag:
+            text = '<' + text
```

Comment:
Shouldn't we just break out of loop here. Since we can assume that to correct the diffs we might at most need only one `<` inserted in the first position.